docs/audit-log-filter-advanced.md

```diff
@@ -0,0 +1,847 @@
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### Conditional Filter Activation
+
+Activate different filters based on conditions:
+
+```json
+{
+  "filter": {
+    "id": "sensitive_operations",
+    "class": {
+      "name": "table_access",
+      "event": [
+        {
+          "name": "insert",
+          "filter": {
+            "class": {
+              "name": "general",
+              "event": [
+                {
+                  "name": "status",
+                  "log": true
+                }
+              ]
+            },
+            "activate": {
+              "field": {
+                "name": "table_name.str",
+                "value": "financial_data"
+              }
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+## Advanced Filtering Patterns
+
+### Account Management Monitoring
+
+Monitor specific account management operations:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "general",
+      "event": [
+        {
+          "name": "status",
+          "print": {
+            "field": {
+              "name": "general_query.str",
+              "print": false,
+              "replace": {
+                "function": {
+                  "name": "query_digest"
+                }
+              }
+            }
+          },
+          "log": {
+            "or": [
+              {
+                "field": {
+                  "name": "general_sql_command.str",
+                  "value": "alter_user"
+                }
+              },
+              {
+                "field": {
+                  "name": "general_sql_command.str",
+                  "value": "create_user"
+                }
+              },
+              {
+                "field": {
+                  "name": "general_sql_command.str",
+                  "value": "drop_user"
+                }
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### Data Modification Tracking
+
+Track only data modification operations:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "table_access",
+      "event": [
+        {
+          "name": "insert",
+          "log": {
+            "not": {
+              "field": {
+                "name": "table_name.str",
+                "value": "audit_log"
+              }
+            }
+          }
+        },
+        {
+          "name": "update",
+          "log": {
+            "not": {
+              "field": {
+                "name": "table_name.str",
+                "value": "audit_log"
+              }
+            }
+          }
+        },
+        {
+          "name": "delete",
+          "log": {
+            "not": {
+              "field": {
+                "name": "table_name.str",
+                "value": "audit_log"
+              }
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### Time-Based Filtering
+
+Use time-based conditions for dynamic filtering:
```

Comment:
There is no "connection_time" field in `connection|xxx` events.
There is `timestamp` but we can filter only by exact match (string representations) of the timestamps.
However, doing this does not make much sense - it will be like I want to filter out all the events that were / will be created at certain point in time.